contexts/ErrorContext.tsx

```diff
@@ -1,14 +1,25 @@
 import { createContext, ReactNode, useState } from "react";
+import { ErrorOriginT } from "types";
+
+type ErrorMessagesT = Record<ErrorOriginT, string[]>;
 
 export interface ErrorContextState {
```

Comment:
Using react nodes for error messages is problematic because then the messages can't be compared directly with each other (they are different objects). I changed to strings because we don't need to pass react nodes anywhere else in the code, and this makes things much simpler.